### PR TITLE
Add support for capital_gains data

### PR DESCRIPTION
**Pull request by thirumalairajr**
Hi,

Adding support for capital_gains for mutual funds & etfs based on the issue https://github.com/ranaroussi/yfinance/issues/926

Result:
image

### CODE REVIEW

**Comment by ValueRaider** [comment]
Sorry I didn't see this in my notifications. This is also fine, except set the base branch to `dev` - it is significantly different from `main` so be prepared for problems.

**Comment by ValueRaider** [comment]
Sorry, this didn't appear in my notifications. Two points otherwise fine:
- is it possible to detect whether ticker is a ETF, so only add capital gains column for that? Something in `info[]` perhaps?
- set base branch to `dev` not `main`, be prepared for conflicts

File: yfinance/base.py [hunk 1]
```diff
@@ -187,6 +186,18 @@ def history(self, period="1mo", interval="1d",
                 proxy = proxy["https"]
             proxy = {"https": proxy}
 
+        #if the ticker is MUTUALFUND or ETF, then get capitalGains events
+        self._get_info(proxy)
+        data = self._info
+        is_capital_gains_data_supported = False
```
Comment (ValueRaider) [inline yfinance/base.py:192]: `is_capital_gains_data_supported` is not necessary, can be removed.

File: yfinance/base.py [hunk 2]
```diff
@@ -187,6 +186,18 @@ def history(self, period="1mo", interval="1d",
...
+        if 'quoteType' in data and data['quoteType'] in ('MUTUALFUND', 'ETF'):
+            is_capital_gains_data_supported = True
+
+        if is_capital_gains_data_supported:
```
Comment (ValueRaider) [inline yfinance/base.py:196]: Combine this if-else with if above

File: yfinance/utils.py [hunk 3]
```diff
@@ -329,9 +329,11 @@ def parse_quotes(data):
     return quotes
 
 
-def parse_actions(data):
+def parse_actions(data, is_capital_gains_data_supported):
```
Comment (ValueRaider) [inline yfinance/utils.py:332]: Remove `is_capital_gains_data_supported`

File: yfinance/utils.py [hunk 4]
```diff
@@ -345,6 +347,14 @@ def parse_actions(data):
 
             dividends.columns = ["Dividends"]
 
+        if is_capital_gains_data_supported and "capitalGains" in data["events"]:
```
Comment (ValueRaider) [inline yfinance/utils.py:350]: Only check if "capitalGains" in data["events"]

File: yfinance/base.py [hunk 5]
```diff
@@ -352,6 +372,13 @@ def history(self, period="1mo", interval="1d",
             df.loc[df["Stock Splits"].isna(), "Stock Splits"] = 0
         else:
             df["Stock Splits"] = 0.0
+        if is_capital_gains_data_supported:
```
Comment (ValueRaider) [inline yfinance/base.py:375]: Instead, check that `capital_gains` is not None and not empty

File: yfinance/base.py (hunk 1 above)
Comment (thirumalairajr) [reply yfinance/base.py:192]: Sure

File: yfinance/base.py (hunk 2 above)
Comment (thirumalairajr) [reply yfinance/base.py:196]: Sure

File: yfinance/utils.py (hunk 3 above)
Comment (thirumalairajr) [reply yfinance/utils.py:332]: Sure

File: yfinance/utils.py (hunk 4 above)
Comment (thirumalairajr) [reply yfinance/utils.py:350]: Sure

File: yfinance/base.py (hunk 5 above)
Comment (thirumalairajr) [reply yfinance/base.py:375]: Sure

File: yfinance/base.py [hunk 6]
```diff
@@ -375,7 +375,8 @@ def history(self, period="1mo", interval="1d",
         df = df[~df.index.duplicated(keep='first')]
         self._history = df.copy()
         if not actions:
-            df = df.drop(columns=["Dividends", "Stock Splits", "Capital Gains"])
+            columns_to_be_dropped = ["Dividends", "Stock Splits", "Capital Gains"]
+            df = df.drop([colname for colname in columns_to_be_dropped if colname in df.columns], axis=1)
```
Comment (ValueRaider) [inline yfinance/base.py:378]: You can set `drop(errors='ignore')` instead. Neater code.

**Comment by ValueRaider** [comment]
`dev` is open now, so when you finish with commits I can merge.

File: README.md [hunk 7]
```diff
@@ -68,6 +68,14 @@ msft.dividends
 # show splits
 msft.splits
 
+
+# show capital gains (for mutual funds & etfs)
+msft.capital_gains
+
+# show financials
+msft.financials
+msft.quarterly_financials
```
Comment (ValueRaider) [inline README.md:77]: Why did you add this?

**Review by ValueRaider — COMMENTED** [review]
One comment

File: README.md (hunk 7 above)
Comment (thirumalairajr) [reply README.md:77]: I will remove these, I think it was added by mistake while resolving merge conflicts

File: yfinance/base.py (hunk 6 above)
Comment (thirumalairajr) [reply yfinance/base.py:378]: Sure will do it

**Comment by thirumalairajr** [comment]
@ValueRaider I have addressed the PR comments
